airsonic-main/src/main/java/org/airsonic/player/service/search/AnalyzerFactory.java

```diff
@@ -1,121 +1,182 @@
+        builder
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\(", "replacement", "", "replace", "all")
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\)$", "replacement", "", "replace", "all")
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\)", "replacement", " ", "replace", "all")
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\{\\}", "replacement", "\\{ \\}", "replace", "all")
+            .addTokenFilter(PatternReplaceFilterFactory.class,
+                    "pattern", "\\[\\]", "replacement", "\\[ \\]", "replace", "all");
+    }
+
+    private Builder createDefaultAnalyzerBuilder() throws IOException {
+        Builder builder = CustomAnalyzer.builder()
+                .withTokenizer(StandardTokenizerFactory.class)
+                .addTokenFilter(CJKWidthFilterFactory.class)
+                .addTokenFilter(ASCIIFoldingFilterFactory.class, "preserveOriginal", "false")
+                .addTokenFilter(LowerCaseFilterFactory.class)
+                .addTokenFilter(StopFilterFactory.class)
+                .addTokenFilter(EnglishPossessiveFilterFactory.class);
+        addTokenFilterForUnderscoreRemovalAroundToken(builder);
+        return builder;
+    }
+
+    private Builder createKeywordAnalyzerBuilder() throws IOException {
+        return CustomAnalyzer.builder()
+                .withTokenizer(KeywordTokenizerFactory.class);
+    }
+
+    private Builder createGenreAnalyzerBuilder() throws IOException {
+        Builder builder = createKeywordAnalyzerBuilder();
+        addTokenFilterForTokenToDomainValue(builder);
+        return builder;
+    }
+
+    /**
+     * Returns the Analyzer to use when generating the index.
+     * 
+     * Whether this analyzer is applied to input values ​​depends on
+     * the definition of the document's fields.
+     * 
+     * @return analyzer for index
+     * @see DocumentFactory
+     */
+    public Analyzer getAnalyzer() {
+        if (isEmpty(analyzer)) {
+            try {
+                analyzer = createDefaultAnalyzerBuilder().build();
+            } catch (IOException e) {
+                LOG.error("Error when initializing Analyzer.", e);
+            }
+        }
+        return analyzer;
+    }
+
+    /**
+     * Returns the analyzer to use when generating a query for index search.
+     * 
+     * String processing handled by QueryFactory
+     * is limited to Lucene's modifier.
+     * 
+     * The processing of the operands is expressed
+     * in the AnalyzerFactory implementation.
+     * Rules for tokenizing/converting input values ​
+     * should not be described in QueryFactory.
+     * 
+     * @return analyzer for query
+     * @see QueryFactory
+     */
+    public Analyzer getQueryAnalyzer() {
+        if (isEmpty(queryAanalyzer)) {
```

Comment:
Typo on `queryAanalyzer`.

Reply:
I revised that according to your advice.